parser/model/ddl.go

```diff
@@ -822,27 +824,52 @@ func (job *Job) IsRollbackable() bool {
 	return true
 }
 
+func (job *Job) IsPausing() bool {
+	return job.State == JobStatePausing
+}
+func (job *Job) IsPaused() bool {
+	return job.State == JobStatePaused
+}
+func (job *Job) IsResuming() bool {
+	return job.IsPaused()
+}
+
+func (job *Job) IsPausable() bool {
+	return (job.NotStarted() || job.IsRunning()) && job.IsRollbackable()
+}
+func (job *Job) IsResumable() bool {
+	return job.IsPaused()
+}
+
 // JobState is for job state.
-type JobState byte
+type JobState int32
 
 // List job states.
 const (
 	JobStateNone    JobState = 0
 	JobStateRunning JobState = 1
+
+	// JobStateRollingback is the state to rollback
 	// When DDL encountered an unrecoverable error at reorganization state,
 	// some keys has been added already, we need to remove them.
-	// JobStateRollingback is the state to do the rolling back job.
 	JobStateRollingback  JobState = 2
 	JobStateRollbackDone JobState = 3
 	JobStateDone         JobState = 4
 	JobStateCancelled    JobState = 5
-	// JobStateSynced is used to mark the information about the completion of this job
-	// has been synchronized to all servers.
+	// JobStateSynced indicates that all operations have been done, and the job
+	// has become archive state.
 	JobStateSynced JobState = 6
 	// JobStateCancelling is used to mark the DDL job is cancelled by the client, but the DDL work hasn't handle it.
 	JobStateCancelling JobState = 7
 	// JobStateQueueing means the job has not yet been started.
 	JobStateQueueing JobState = 8
+
+	JobStatePaused  JobState = 9
+	JobStatePausing JobState = 10
+)
+
+const (
+	JobStateCheckInterval = 50 * time.Millisecond
```

Comment:
Whether the 50 milliseconds checkInterval is too frequent?  can we set it 500 milliseconds?

Reply:
not necessary